### PR TITLE
DOC: Add .zenodo.json for author association

### PR DESCRIPTION
**Pull request by thewtex**
Based on https://github.com/nipy/nipype/blob/8fe05afcd7c716b52044837c96749ef420154fbb/.zenodo.json

This mechanism explicitly associates authors with their
[ORCID](https://orcid.org/).

This commit seeds the configuration. Additional authors should
self-identify by adding their ORCID to this file.

h/t @satra

We may want to add a note to *CONTRIBUTING.md* after merged.

### CODE REVIEW

**Comment by thewtex** [comment]
After initialization, I will post to Discourse requesting contributors add their ORCID.

**Comment by satra** [comment]
@thewtex - in nipype we have some custom auto inclusion based on commits + management.

https://github.com/nipy/nipype/blob/master/tools/update_zenodo.py

just a note that no system of contributors for a large system is good enough for everyone !

**Comment by thewtex** [comment]
@satra awesome! I'll try the script...

**Comment by phcerdan** [comment]
This is gold, thanks @thewtex.

A related question, does somebody know what is the criteria of Google Scholar to parse Zenodo? It seems that some DOI's are integrated, but not all of them.

File: .zenodo.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "creators": [
+    {
+      "affiliation": "Kitware, Inc",
+      "name": "McCormick, Matthew",
+      "orcid": "0000-0001-9475-3756"
+    }
+  ],
```
Comment (hjmjohnson) [inline .zenodo.json]:
```
    {
      "affiliation": "University of Iowa",
      "name": "Johnson, Hans",
      "orcid": "0000-0001-9513-2660"
    }
```

**Comment by jhlegarreta** [comment]
Awesome @thewtex !

Adding a note to [CONTRIBUTING.md](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md) looks fair. I think Nibabel has already some comment about it in their [developper's guide](https://github.com/nipy/nibabel/blob/master/.github/CONTRIBUTING.md), but the website seems to be down now.

To complete what @satra has said, a few related thoughts in an earlier issue: 
https://github.com/InsightSoftwareConsortium/ITK/issues/456#issuecomment-476200207

If the script to parse authors based on the commits+management is finally found to fit our needs, we may want to add it to the [Utilities/Maintenance](https://github.com/InsightSoftwareConsortium/ITK/tree/master/Utilities/Maintenance) folder, and add a section or note to the [Release.md](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Documentation/Maintenance/Release.md) document.

> A related question, does somebody know what is the criteria of Google Scholar to parse Zenodo? It seems that some DOI's are integrated, but not all of them.

No idea @phcerdan. Sorry.

File: .zenodo.json (hunk 1 above)
Comment (jhlegarreta) [reply .zenodo.json]: {
      "affiliation": "Universit\u00e9 de Sherbrooke",
      "name": "Legarreta, Jon Haitz",
      "orcid": "0000-0002-9661-1396"
    }

Not sure whether we are willing to separate the original *creators* of the toolkit vs. *contributors*. It is apparently possible according to the zenodo documentation: https://developers.zenodo.org/, but the *contributors* tag description does not fully match what a contributor is in terms of software/the toolkit.

**Comment by jhlegarreta** [comment]
> Adding a note to [CONTRIBUTING.md](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md) looks fair. I think Nibabel has already some comment about it in their [developper's guide](https://github.com/nipy/nibabel/blob/master/.github/CONTRIBUTING.md), but the website seems to be down now.

Ah, sorry, it was [Nypipe](https://nipy.org/packages/nipype/index.html) that had a complete web page and details about how to cite the toolkit and inviting developers to open a PR to modify the _.zenodo.json_ file with their data when contributing for the first time:
https://nipype.readthedocs.io/en/latest/about.html

They also cross-reference the GitHub contributors section.

At the end of the page, they also have a section dedicated to acknowledging funding parties: might be an inspiration for #454.

**Comment by hjmjohnson** [comment]
@thewtex FYI:  These errors are similar to the ones I was encountering on a previous version of the #1471 branch yesterday.

File: .zenodo.json (hunk 1 above)
Comment (thewtex) [reply .zenodo.json]: >  "orcid": "0000-0001-9513-2660"

>  "orcid": "0000-0002-9661-1396"

@hjmjohnson got these added.

>  contributors tag description does not fully match what a contributor is in terms of software/the toolkit.

Yes, *contributors* seems more appropriate for a small, confined project where people are full time with dedicated roles as described versus a community-driven project like ITK.

**Comment by thewtex** [comment]
> https://github.com/nipy/nipype/blob/master/tools/update_zenodo.py

I added a script based on this and a version in nibabel @jhlegarreta referenced. Running this script is documented in the release process notes so .zenodo.json is automatically updated.

> Adding a note to CONTRIBUTING.md looks fair.

Done.

> we may want to add it to the Utilities/Maintenance folder, and add a section or note to the Release.md document.

Done.

> These errors are similar to the ones I was encountering on a previous version of the #1471 branch yesterday.

The source of this is not clear or consistent -- I will exclude them from the CI to avoid false positives.

File: .zenodo.json [hunk 2]
```diff
@@ -4,6 +4,946 @@
...
+      "name" : "Francois Budin"
+    },
+    {
+      "name" : "Will Schroeder"
+    },
+    {
+      "name" : "Stephen R. Aylward"
+    },
+    {
+      "name" : "Dženan Zukić"
```
Comment (dzenanz) [inline .zenodo.json]: Is now the right time to add ORCIDs (mine is 0000-0002-9546-9703), or in follow-up PRs?

**Comment by hjmjohnson** [comment]
FYI:  https://orcid.org/orcid-search/quick-search/?searchQuery=Luis%20Ibanez

| ORCID | First | Last | _ | Affiliations |
| -------|------|-----|--|------------|
| https://orcid.org/0000-0003-2712-8584 | Luis | Ibanez |   | Google Inc, Kitware Inc, University of North Carolina at Chapel Hill, Université de Rennes 1, Universidad Industrial de Santander |
| https://orcid.org/0000-0002-9546-9703 | Dženan | Zukić |   | Kitware Inc, University of Siegen |
|https://orcid.org/0000-0002-6498-6273 | Daniel | Blezek |   | Mayo Clinic, Mayo Clinic Graduate School for Biomedical Sciences |
| https://orcid.org/0000-0002-9020-3770 | Julien | Jomier |   | Kitware SAS, Kitware Inc |
| https://orcid.org/0000-0002-7499-3514 | Lydia | Ng |   | Allen Institute for Brain Science |
|https://orcid.org/0000-0002-7862-8856 | Stephen | Aylward | | |
| https://orcid.org/0000-0002-4212-3362 | Brian | Avants | | |
| https://orcid.org/0000-0001-9418-5103 | NICHOLAS | TUSTISON |   | University of California Irvine, University of Virginia, Washington University in Saint Louis, Brigham Young University |
| https://orcid.org/0000-0002-9688-8950 | Jean-Christophe | Fillion-Robin |   | Kitware Inc, École Polytechnique Fédérale de Lausanne, Ecole Supérieure Chimie Physique Electronique de Lyon |
| https://orcid.org/0000-0001-8639-5354 | Vincent | Magnotta | | |
| https://orcid.org/0000-0001-8543-4016 | Paul | Yushkevich |   | University of Pennsylvania | |
| https://orcid.org/0000-0002-8858-1254 | Ali | Ghayoor | 
|

**Comment by hjmjohnson** [comment]
Just an idea, but for credit in the zendo file, should we have a "long-tail truncation" where at least N (2, 3, 10, 50) commits are necessary in order to be listed?   I'm always concerned that a long tail listing dilutes the benefits of being listed.   My knee jerk reaction would be some number that includes approximately the top 50 developers.  Just an idea.

File: .zenodo.json (hunk 2 above)
Comment (hjmjohnson) [reply .zenodo.json]: I think they should be added now, and merged soon.

**Comment by jhlegarreta** [comment]
> Just an idea, but for credit in the zendo file, should we have a "long-tail truncation" where at least N (2, 3, 10, 50) commits are necessary in order to be listed? I'm always concerned that a long tail listing dilutes the benefits of being listed. My knee jerk reaction would be some number that includes approximately the top 50 developers. Just an idea.

I am in two minds about this. I think giving credit to all contributors, regardless of the number of commits, makes the community a more welcoming place. GitHub lists all contributors. and the list can be ordered in different ways, always showing the top contributors by number of commits or additions/deletions. And there are other sites, if I am not mistaken, that list the top contributors according to different criteria.

Remote module developers would not be credited by the current script either. May be this is also controversial, since they would not actually appear as having made a commit to ITK itself. Not sure if when the remote makes into ITK proper, the remote's history also makes it into ITK. If not, I assume they would not get their information into zenodo.

Sorry Hans, more questions than answers.

**Comment by phcerdan** [comment]
> Just an idea, but for credit in the zendo file, should we have a "long-tail truncation" where at least N (2, 3, 10, 50) commits are necessary in order to be listed? I'm always concerned that a long tail listing dilutes the benefits of being listed. My knee jerk reaction would be some number that includes approximately the top 50 developers. Just an idea.

I agree with Hans, long-tails dilutes the benefits indeed, NCommits>70(?), or NModules >= 1?

File: .zenodo.json (hunk 2 above)
Comment (thewtex) [reply .zenodo.json]: > I think they should be added now, and merged soon.

@hjmjohnson could you please push the id's you provided below to the branch?

**Comment by thewtex** [comment]
> Just an idea, but for credit in the zendo file, should we have a "long-tail truncation" where at least N (2, 3, 10, 50) commits are necessary in order to be listed?

In d9a3ede a requirement for 10 commits is added. I think 50 is too high -- at 10, a contributor is often also contributing code reviews. This may also be helpful in reducing the burden for new contributors by not need to worry about adding an affiliation and an ORCID iD to .zenodo.json.

> Remote module developers would not be credited by the current script either.

This is ok -- remote modules have their own repository, which can have a Zenodo record.

File: .zenodo.json [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+      "name": "Robert Tamburo"
+    },
+    {
+      "name": "Michel Audette"
+    },
+    {
+      "name": "Mark Foskey"
+    },
+    {
+      "name": "Pablo Hernandez-Cerdan"
```
Comment (phcerdan) [inline .zenodo.json]:
```suggestion
      "name": "Hernandez-Cerdan, Pablo"
      "orcid":  "0000-0002-1689-3451"
```

Comment (thewtex) [reply .zenodo.json]: Good catch -- the commas were missing from the initial addition, and they were missing from the script. Added in the latest push.

**Comment by thewtex** [comment]
Merging -- @hjmjohnson if you create a separate PR for the ORCID's found, the contributors can review them.

**Pull request merged**